src/utils/fetch.ts

```diff
@@ -1,36 +1,11 @@
 ﻿import log from '../log';
-
-export type HTTPMethodType =
-    | 'get'
-    | 'put'
-    | 'post'
-    | 'delete'
-    | 'patch'
-    | 'options'
-    | 'head';
-
-export interface OAPICallResult {
-    response?: any;
-    status: number;
-    headers: Headers;
-    size: number;
-    url: string;
-    responseType?: string;
-    isNetworkError?: never;
-}
-
-export interface NetworkError {
-    message?: string | Error;
-    isNetworkError: true;
-    status?: never;
-}
+import type { HTTPMethodType, NetworkError, OAPIRequestResult } from '../types';
 
 interface Options {
     /**
-     *  The body of the request. If this is an object, that is not already handled by the body mixin
-     *  it is converted to JSON and the appropriate content-type header added.
+     *  The body of the request.
      */
-    body?: BodyInit | Record<string, unknown>;
+    body?: any;

```

Comment:
do we really allow anything?

Reply:
Theoretically yes, our fetch wrapper passes everything not compatible with BodyInit (type expected by original fetch) through `JSON.stringify` which accepts anything and returns `string|undefined`.